builtin/game/minimap.lua

```diff
@@ -0,0 +1,69 @@
+local minimap_def = {
+	hud_elem_type = "minimap",
+	position = {x=1, y=0},
+	alignment = {x=-1, y=1},
+	offset = {x=-10, y=10},
+	size = {x = 256 , y = 256}
+}
+
+local minimap_hud_id = {}
+local function update_builtin_minimap(player)
+	local name = player:get_player_name()
+	local id = minimap_hud_id[name]
+	-- mimic wired flag behavior of deprecated non HUD element minimap
+	if player:hud_get_flags().minimap then
+		if not id then
+			minimap_hud_id[name] = player:hud_add(minimap_def)
+		end
+	else
+		if id then
+			player:hud_remove(id)
+		end
+		minimap_hud_id[name] = nil
+	end
+end
+
+core.register_on_mods_loaded(function()
+	core.register_on_joinplayer(function(player)
+		update_builtin_minimap(player)
+	end)
+end)
+
+local function player_event_handler(player, eventname)
+	assert(player:is_player())
+	local name = player:get_player_name()
+	if name == "" then
```

Comment:
How can this happen?

Reply:
I adopted this from `builtin/game/statbars.lua`, so I guess `player_event_handler` may not get a player object.
(The Lua API documentation says: `get_player_name(): Returns player name or "" if is not a player`)